src/routes/write-exercise/components/Feedback.tsx

```diff
@@ -59,11 +59,11 @@ const Feedback = ({ documentWithResult, submission, needsToBeRepeated }: Feedbac
   } else {
     Icon = CloseCircleIconBold
     background = BannerRed
-    message = needsToBeRepeated
-      ? `${getLabels().exercises.write.feedback.wrong}\n${wrongWithCorrectSolution}`
-      : wrongWithCorrectSolution
+    message =
+      needsToBeRepeated && !submission
+        ? wrongWithCorrectSolution
```

Comment:
Could you please rename `wrongWtihCorrectSolution` to `correctSolutionHint` in L43 and the label `exercises.write.feedback.wrongWithSolution` in labels.json L78 to `solution`? Thanks :)